### ISSUE TITLE
Reproducing resnet-200 results

### ISSUE THREAD

**Issue by dubur** [facebook/fb.resnet.torch#47]
I download the resnet 200 model and test it on ILSVRC2012 validation set and got similar top1 and top5 on 224x224 scale, but when I change the scale to 320, I got 20.5(top1) and 5.2(top5).

The main problem should be the size of feature map. I wonder how to process the feature map right after the global average pooling layer, because 7x7 pooling will get larger feature maps (2048x4x4) for 320x320 input, is it right to take the average of a 4x4 map and turn it into 1x1?

**Comment by colesbury** [comment]
You apply the fully-connected layer in as a convolution. Basically you replace the nn.Linear with a 1x1 SpatialConvolution with the same weights and biases. This will give you a 1000x4x4 tensor. You then average over the 4x4 part. I find you get slightly better accuracy if you exp(output) before averaging, since the network predicts log-probabilities.

I'll try to add the code to compute this soon.

**Comment by dubur** [comment]
Thanks ! @colesbury I guess it is a standard way to handle enlarged input scales for neural network

**Comment by MissT157** [comment]
So, for a different size of an image, should I just change the kernel size of a pooling layer? or there is more to it?

**Comment by dubur** [comment]
@MissT157  You should keep the model structure the same before fully-connected layer and change the last fc layer into an equivalent convolution layer, the details of how to change the layer has been explained up there
